---
title: "Differential Expression Analysis"
output:
  html_document:
    df_print: paged
---

## Install libraries

```{r}
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("DESeq2")

#install.packages("pheatmap")
```


## Load libraries

```{r}

library(DESeq2)
library(pheatmap)
```

## Load count matrix

```{r}
x = read.table("sample.counts", row.names=1, header=T, sep=",")
s = read.table("sample.info", header=T, row.names=1, colClasses=c("character", "factor"))
```

## Create DESeq2 object

```{r}
dds = DESeqDataSetFromMatrix(countData = x, colData = s, design = ~ condition)
```

## Differential expression analysis

Run a differential expression analysis (Tumour vs. Normal) using a log-fold change threshold of 1

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#differential-expression-analysis

```{r}
dds <- DESeq(dds)
res <- results(dds, lfcThreshold=1)
res
```

## MA-plot

Generate an MA-plot

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#exploring-and-exporting-results
```{r}
plotMA(res, ylim=c(-10,10))
```
## Plot the normalized counts for the GJB2 gene

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#plot-counts

```{r}
plotCounts(dds, gene='GJB2', intgroup="condition")
```

## PCA

Generate a PCA plot of the samples using the transformed count data

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#extracting-transformed-values

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#principal-component-plot-of-the-samples

```{r}
vsd <- vst(dds, blind=FALSE)
plotPCA(vsd, intgroup=c("condition"))
```

## Heatmap

Visualize the differential gene expression results as a heatmap.

Take the top 20 genes according to the adjusted p-value

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#heatmap-of-the-count-matrix

```{r}
top20 <- row.names(head(res[order(res$padj, decreasing = FALSE),], n=20))
pheatmap(assay(vsd)[top20,], cluster_rows=TRUE, show_rownames=TRUE,cluster_cols=TRUE)
```

## Export

Export the significant results (padj < 0.01) to a CSV file

Tutorial: http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#exporting-results-to-csv-files
```{r}
resOrdered <- res[order(res$pvalue),]
resSig <- subset(resOrdered, padj < 0.1)
resSig
write.csv(as.data.frame(resSig), 
          file="sig_results.csv")
```
